pkg/terraform/stages/gcp/stages.go

```diff
@@ -118,20 +120,21 @@ func extractGCPLBConfig(s stages.SplitStage, directory string, terraformDir stri
 		return "", err
 	}
 
-	// Update the ignition bootstrap variable to include the lbconfig.
-	tfvarData["ignition_bootstrap"] = string(ignitionOutput)
+	clusterID, ok := tfvarData["cluster_id"]
+	if !ok {
+		return "", fmt.Errorf("failed to read cluster id from tfvars")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*1)
+	defer cancel()
 
-	// Convert the bootstrap data and write the data back to a file. This will overwrite the original tfvars file.
-	jsonBootstrap, err := json.Marshal(tfvarData)
+	bucketName := gcpbootstrap.GetBootstrapStorageName(clusterID.(string))
+	bucketHandle, err := gcpbootstrap.CreateBucketHandle(ctx, bucketName)
 	if err != nil {
-		return "", fmt.Errorf("failed to convert bootstrap ignition to bytes: %w", err)
+		return "", fmt.Errorf("failed to create bucket handle %s: %w", bucketName, err)
 	}
-	tfvarsFile.Data = jsonBootstrap
-
-	// update the value on disk to match
-	if err := os.WriteFile(fmt.Sprintf("%s/%s", directory, tfvarsFile.Filename), jsonBootstrap, 0o600); err != nil {
-		return "", fmt.Errorf("failed to rewrite %s: %w", tfvarsFile.Filename, err)
+	if err := gcpbootstrap.FillBucket(context.Background(), bucketHandle, string(ignitionOutput)); err != nil {
+		return "", fmt.Errorf("failed to fill gcp bucket with updated boostrap ignition contents: %w", err)
```

Comment:
I still think it's confusing to fill the GCP bucket here as this function is intended for another purpose (custom DNS). I'm a little confused how this seems to be working, because earlier in the function there is:

```
if !userConfiguredDNSRaw.(bool) {
		return "", nil
	}
```
which should bail out if we're not using custom DNS.

In any case, it still seems better to me to keep the uploading of ignition in Terraform and not combine it with this custom DNS flow.

Reply:
@patrickdillon The bucket is only re-filled in this case. The bucket is originally filled in tfvars with the original data. Do we still want to use terraform as shown below?

Reply:
Got it! That works well. Sorry i missed it